catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.graphics.PointF;
+
+import java.util.ArrayList;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class EmbroideryManager {
```

Comment:
im not a huge fan of "Manager" classes. that either means its doing way too much, or its not behaving as what it actually is responsible for.

anyways, this class here only holds the list and a lock, and is more or less acting just as a proxy to the list. so why not go there and make it a list?
have this extend list, and overrite add, clear, etc.

whats also a bit odd is that for adding and clearing this handles the lock itself, but for iterating you have to deal with this on the outside. so either all in, or all out.
id prefer all in, if you overwrite this as a list you can also overwrite listIterator and deal with the locking there.

on the other hand, there are thread safe collections in java already that you could extend upon, instead of list and dealing with locking yourself...
https://stackoverflow.com/questions/8203864/choosing-the-best-concurrency-list-in-java